apps/daimo-mobile/src/view/screen/history/shared.ts

```diff
@@ -11,34 +11,58 @@ import { env } from "../../../env";
 // Either uses the memo field for standard transfers, e.g. "for ice cream"
 // Or generates a synthetic one for swaps, e.g. "5 USDT -> USDC" if short
 // or "Accepted 5 USDT as USDC" if long
-export function getSynthesizedMemo(
-  op: DisplayOpEvent,
-  daimoChain: DaimoChain,
-  short?: boolean
-) {
+export function getSynthesizedMemo({
+  op,
+  daimoChain,
+  short,
+  sentByUs,
+}: {
+  op: TransferClog;
+  daimoChain: DaimoChain;
+  short?: boolean;
+  sentByUs?: boolean;
+}) {
   const chainConfig = env(daimoChain).chainConfig;
   const coinName = chainConfig.tokenSymbol.toUpperCase();
 
   if (op.memo) return op.memo;
   if (op.type === "createLink" && op.noteStatus.memo) return op.noteStatus.memo;
   if (op.type === "claimLink" && op.noteStatus.memo) return op.noteStatus.memo;
 
-  if (op.type !== "transfer") return null;
-  if (op.requestStatus) {
+  if (op.type !== "transfer" && op.type !== "swap") return null;
+  if (op.type === "transfer" && op.requestStatus) {
     return op.requestStatus.memo;
-  } else if (op.preSwapTransfer) {
-    if (isNativeETH(op.preSwapTransfer.coin.address, chainConfig)) {
-      return `ETH → ${coinName}`;
+  } else if (
+    (op.type === "transfer" && op.preSwapTransfer) ||
+    op.type === "swap"
+  ) {
+    const isOutboundSwap = !!(op.type === "swap" && sentByUs);
+
+    // TODO: do we need to explicitly handle preSwapTransfer && "transfer" for
+    // backwards compatibility?
+    const otherCoin =
```

Comment:
@dcposch do we need to keep preSwapTransfer here or can I get rid of it entirely (aka are ops cached on device in any way that could break)

Reply:
they're on device, yes--now that we have `storedAccount/storedTypes.ts`, you can always check whether something's there plus it's typechecked